src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestCallback.cs

```diff
@@ -56,6 +60,14 @@ private static void RequestCallback(
             {
                 switch (internetStatus)
                 {
+                    case Interop.WinHttp.WINHTTP_CALLBACK_STATUS_CONNECTED_TO_SERVER:
+                        if (WinHttpHandler.CertificateCachingAppContextSwitchEnabled)
+                        {
+                            IPAddress connectedToIPAddress = IPAddress.Parse(Marshal.PtrToStringUni(statusInformation)!);
```

Comment:
Is it possible to use `IPAddress.Parse(ReadOnlySpan<char>)` here? Moreover: are we confident this will work well for both IPv4 and IPv6?

Reply:
> Moreover: are we confident this will work well for both IPv4 and IPv6?

IIRC, this was working perfectly fine both over v4 v6.

> Is it possible to use IPAddress.Parse(ReadOnlySpan<char>) here?

I can take a look.

Reply:
> Is it possible to use IPAddress.Parse(ReadOnlySpan) here?

No, we can't use it because it doesn't exist on framework